core/src/main/scala/kafka/server/ControllerApis.scala

```diff
@@ -229,9 +229,9 @@ class ControllerApis(
     // Check if topic deletion is enabled at all.
     if (!config.deleteTopicEnable) {
       if (apiVersion < 3) {
-        throw new InvalidRequestException("Topic deletion is disabled.")
+        return CompletableFuture.failedFuture(new InvalidRequestException("This version does not support topic deletion."))
```

Comment:
nit
```suggestion
        return CompletableFuture.failedFuture(new InvalidRequestException("This version" + apiVersion + "does not support topic deletion."))
```

Reply:
If we want to add `apiVersion` here, how about changing "This" to "The"?

Reply:
I'm fine with it